apps/institution-service/src/middlewares/validationMiddleware.ts

```diff
@@ -0,0 +1,91 @@
+import { UiUserPermissions } from "@repo/shared-utils";
+import { NextFunction, Request, Response } from "express";
+import Joi, { ObjectSchema } from "joi";
+import jwt from "jsonwebtoken";
+import { Institution } from "../models/institution";
+
+export const validate = (schema: ObjectSchema) => {
+  return (req: Request, res: Response, next: NextFunction) => {
+    const { error } = schema.validate(req.body);
+
+    if (error) {
+      return res.status(400).json({ error: error.details[0].message });
+    }
+
+    next();
+  };
+};
+
+export const institutionUpdateSchema = Joi.object({
+  name: Joi.string(),
+  keywords: Joi.string(),
+  logo: Joi.string()
+    .uri({ scheme: ["http", "https"] })
+    .messages({
+      "string.uri": "The url must be a valid URL",
+    }),
+  url: Joi.string()
+    .uri({ scheme: ["http", "https"] })
+    .messages({
+      "string.uri": "The url must be a valid URL",
+    }),
+  is_test_bank: Joi.boolean(),
+  routing_numbers: Joi.array()
+    .items(Joi.string().pattern(/^\d{9}$/, "9 digits"))
+    .messages({
+      "string.pattern.base": "Each routing number must be exactly 9 digits",
+      "array.base": "Routing numbers must be an array of strings",
+    }),
+});
+
+export const institutionCreateSchema = institutionUpdateSchema.append({
+  ucp_id: Joi.string().required(),
```

Comment:
Hmm, these database fields are all snake case, should we refactor those fields? I think it will require an update here and on ucw-app. It's probably a big enough task to create 2 tickets, one for each repo.

Reply:
We can create tickets to do it later

Reply:
Sounds good I made this ticket https://universalconnect.atlassian.net/browse/UCP-369